src/annotator/anchoring/pdf.ts

```diff
@@ -259,6 +259,27 @@ function isSpace(char: string) {
 
 const isNotSpace = (char: string) => !isSpace(char);
 
+/**
+ * Determines if provided text layer is done rendering.
+ * It works on older PDF.js versions which expose a public renderingDone prop,
+ * and newer versions as well
+ */
+export function isTextLayerRenderingDone(textLayer: TextLayer): boolean {
+  if (textLayer.renderingDone !== undefined) {
+    return textLayer.renderingDone;
+  }
+
+  if (!textLayer.div) {
+    return false;
+  }
+
+  // When a Page is rendered, the div gets an element with the class
+  // endOfContent appended to it. If that element exists, we can consider the
+  // text layer is done rendering.
+  // See https://github.com/mozilla/pdf.js/blob/1ab9ab67eed886f27127bd801bc349949af5054e/web/text_layer_builder.js#L103-L107
+  return textLayer.div.querySelector('.endOfContent') !== null;
```

Comment:
This queries for _any_ descendant matching the class. Should it have used a more conservative `hasClass` check?

Reply:
The class is not set to the div itself, but to a direct children. That's why I used `querySelector`.

Reply:
Ah sorry, I skimmed the upstream diff too quickly.